### ISSUE TITLE
PostAuthor component can cause a crash for some users

### ISSUE THREAD

**Issue by notnownikki** [WordPress/gutenberg#4437]
The PostAuthor component references `user.capabilities.level_1`, and `user.capabilities` is sometimes undefined. This causes an immediate crash.

I don't know exactly the circumstances that cause `user.capabilities` to be undefined, but we had this problem a while back in #2702 and it was fixed for most cases in #4122 .

The check occurs in `editor/components/post-author/index.js` line 42.

Is there a check we can replace this with, in the same way as we did in #4122?

If someone could give me some insight as to what we should be checking here, I'd quite like to implement the fix so I learn more about this :)

**Comment by Kilian** [comment]
I have this problem still with gutenberg 2.0.0 (Hoped #2702 would fix it, as that is the error I'm getting), not running multisite, single author/admin (me), but has been running + upgraded since WP 2.0. I did a check for a potentially problematic plugins (wp super cache) but disabling that didn't help either.

**Comment by notnownikki** [comment]
Looking at the change that fixed it for most people, we need to make sure the component gets the new `post_capabilities` data from the API, and that's been done for most components but not the `PostAuthor` component.

I think I have enough info to start working on a fix now.

**Comment by notnownikki** [comment]
Ooooookay, it's not quite the bug I thought, so updating here in case anyone sees this and goes, "oh yes, just flip this switch and all is fine.".

The component gets all the users in an edit context, and checks if they have at least `level_1` capabilities. However, if you're a user that doesn't have an explicit role on a site, then you don't have these capabilities listed, and when the component tries to render you in the dropdown list of potential authors, you get the crash.

This can't be fixed in the same way as the other fixes because it's using a different API, so I'm not sure on what the fix would be yet. Investigating further.

**Comment by Kilian** [comment]
Have you seen https://core.trac.wordpress.org/ticket/16841 ? That might provide some additional context/info!

**Comment by notnownikki** [comment]
That could explain why you're getting the crash! My situation is different, but results in the same error. I'm currently working out if it's one particular user (which would point to the bug you linked, or similar), or if I'm not getting any capabilities back from the API at all. Will keep this issue updated!

**Comment by MarkCanada** [comment]
I have that same problem on my test system (WP 4.9.4, GB 2.3.0, TwentySixteen Theme). I disabled all plugins but Gutenberg and still can't edit anything.
For some reason in the user manager it displays under additonal capabilities "frm_view_forms, frm_edit_forms, frm_delete_forms, frm_change_settings, frm_view_entries, frm_delete_entries" so I added another admin user account and used a user switcher plugin and that user has the same problem.
> TypeError: Cannot read property 'level_1' of undefined
